src/events/redux/middleware.js

```diff
@@ -151,8 +151,9 @@ export default function eventsMiddleware(store) {
         ).then(logs => {
           const swapTxIds = _.map(eventSelectors.getFetchedSwapFills(store.getState()), 'transactionHash')
           const newSwapFills = _.filter(logs, ({ transactionHash }) => !_.includes(swapTxIds, transactionHash))
-          if (logs && logs.length) {
-            store.dispatch(makeEventFetchingActionsCreators('swapFills').got(newSwapFills))
+          if (logs && logs.length && newSwapFills.length) {
```

Comment:
You might also need to null check newSwapFills here?

Reply:
_.filter(undefined) returns an array, so it should be safe